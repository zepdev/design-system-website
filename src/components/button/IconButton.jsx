import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import ButtonBase from './ButtonBase'

function IconButton({ classes, className: classNameProp, children: childrenProp, ...other }) {
  const children = React.cloneElement(childrenProp, {
    className: classnames('zep-button__icon', classes.icon),
  })
  return (
    <ButtonBase
      data-testid="iconButton"
      className="zep-button zep-button-primary zep-button-icon"
      {...other}
    >
      {children}
    </ButtonBase>
  )
}

IconButton.propTypes = {
  className: PropTypes.object,
  children: PropTypes.object.isRequired,
  classes: PropTypes.object,
}

export default IconButton
