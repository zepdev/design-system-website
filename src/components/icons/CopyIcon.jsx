import React from 'react'
import PropTypes from 'prop-types'

const CopyIcon = ({ className, width, height, fill, viewBox }) => (
  <svg
    width={width}
    height={height}
    xmlns="http://www.w3.org/2000/svg"
    viewBox={viewBox}
    aria-labelledby="title"
    className={className}
  >
    <title id="title">Copy Icon</title>
    <path fill="none" d="M0 0h24v24H0V0z"/>
    <path fill={fill} d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4H8c-1.1 0-1.99.9-1.99 2L6 21c0 1.1.89 2 1.99 2H19c1.1 0 2-.9 2-2V11l-6-6zM8 21V7h6v5h5v9H8z"/>
  </svg>
)

CopyIcon.propTypes = {
  width: PropTypes.string,
  height: PropTypes.string,
  viewBox: PropTypes.string,
  fill: PropTypes.string,
}

CopyIcon.defaultProps = {
  width: '32',
  height: '32',
  viewBox: '0 0 24 24',
  fill: 'currentColor',
}

export default CopyIcon
