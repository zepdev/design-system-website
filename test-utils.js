import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@material-ui/core/styles'
import theme from './src/theme'

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
