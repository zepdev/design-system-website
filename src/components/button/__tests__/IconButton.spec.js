import React from 'react'
import { render } from '../../../../test-utils'
import IconButton from '../IconButton'
import CopyIcon from '../../icons/CopyIcon'

describe('IconButton', () => {
  it('renders correctly', () => {
    const mockProps = {
      onClick: jest.fn(() => {}),
      children: <CopyIcon className="test" />,
    }
    const component = render(<IconButton {...mockProps} />)
    expect(component).toMatchSnapshot()
  })
})
