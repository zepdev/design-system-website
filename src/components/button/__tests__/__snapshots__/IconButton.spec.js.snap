// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="zep-button zep-button--icon makeStyles-button-1 zep-button--primary"
      >
        <svg
          aria-hidden="true"
          class="zep-button__icon"
          data-testid="ZepIconEssentialStarIcon"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M11.85 2.01a1.3 1.3 0 00-1.01.7L8.2 7.94l-5.03.5c-.5.04-.93.37-1.1.85l-.04.13c-.11.44.02.9.35 1.23l4.16 4.12L5 20.36c-.13.5.04 1.02.44 1.35l.11.08c.39.25.88.28 1.3.07L12 19.31l5.16 2.56c.46.22 1 .16 1.4-.16l.1-.1c.33-.32.46-.8.34-1.24l-1.54-5.6 4.16-4.13c.36-.35.48-.88.3-1.36l-.05-.12c-.2-.41-.6-.69-1.06-.73l-5.01-.5-2.63-5.21A1.3 1.3 0 0012 2l-.15.01zM12 3.25c.02 0 .04.02.05.03l2.79 5.53c.1.19.28.32.5.34l5.36.53c.03 0 .04.01.05.03l-.01.05-4.42 4.37a.62.62 0 00-.16.61l1.64 5.95c0 .02 0 .04-.02.05a.05.05 0 01-.06 0l-5.44-2.69a.62.62 0 00-.56 0l-5.43 2.7-.06-.01a.05.05 0 01-.02-.06l1.64-5.94a.62.62 0 00-.17-.6l-4.42-4.4a.05.05 0 01.03-.07l5.38-.54c.21-.02.4-.15.5-.34l2.78-5.52c0-.01.03-.03.05-.03z"
            fill="currentColor"
            fill-rule="evenodd"
            transform="scale(Infinity, Infinity)"
          />
        </svg>
      </button>
      <button
        class="zep-button zep-button--icon zep-button--secondary"
      >
        <svg
          aria-hidden="true"
          class="zep-button__icon"
          data-testid="ZepIconEssentialStarIcon"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M11.85 2.01a1.3 1.3 0 00-1.01.7L8.2 7.94l-5.03.5c-.5.04-.93.37-1.1.85l-.04.13c-.11.44.02.9.35 1.23l4.16 4.12L5 20.36c-.13.5.04 1.02.44 1.35l.11.08c.39.25.88.28 1.3.07L12 19.31l5.16 2.56c.46.22 1 .16 1.4-.16l.1-.1c.33-.32.46-.8.34-1.24l-1.54-5.6 4.16-4.13c.36-.35.48-.88.3-1.36l-.05-.12c-.2-.41-.6-.69-1.06-.73l-5.01-.5-2.63-5.21A1.3 1.3 0 0012 2l-.15.01zM12 3.25c.02 0 .04.02.05.03l2.79 5.53c.1.19.28.32.5.34l5.36.53c.03 0 .04.01.05.03l-.01.05-4.42 4.37a.62.62 0 00-.16.61l1.64 5.95c0 .02 0 .04-.02.05a.05.05 0 01-.06 0l-5.44-2.69a.62.62 0 00-.56 0l-5.43 2.7-.06-.01a.05.05 0 01-.02-.06l1.64-5.94a.62.62 0 00-.17-.6l-4.42-4.4a.05.05 0 01.03-.07l5.38-.54c.21-.02.4-.15.5-.34l2.78-5.52c0-.01.03-.03.05-.03z"
            fill="currentColor"
            fill-rule="evenodd"
            transform="scale(Infinity, Infinity)"
          />
        </svg>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="zep-button zep-button--icon makeStyles-button-1 zep-button--primary"
    >
      <svg
        aria-hidden="true"
        class="zep-button__icon"
        data-testid="ZepIconEssentialStarIcon"
        focusable="false"
        height="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M11.85 2.01a1.3 1.3 0 00-1.01.7L8.2 7.94l-5.03.5c-.5.04-.93.37-1.1.85l-.04.13c-.11.44.02.9.35 1.23l4.16 4.12L5 20.36c-.13.5.04 1.02.44 1.35l.11.08c.39.25.88.28 1.3.07L12 19.31l5.16 2.56c.46.22 1 .16 1.4-.16l.1-.1c.33-.32.46-.8.34-1.24l-1.54-5.6 4.16-4.13c.36-.35.48-.88.3-1.36l-.05-.12c-.2-.41-.6-.69-1.06-.73l-5.01-.5-2.63-5.21A1.3 1.3 0 0012 2l-.15.01zM12 3.25c.02 0 .04.02.05.03l2.79 5.53c.1.19.28.32.5.34l5.36.53c.03 0 .04.01.05.03l-.01.05-4.42 4.37a.62.62 0 00-.16.61l1.64 5.95c0 .02 0 .04-.02.05a.05.05 0 01-.06 0l-5.44-2.69a.62.62 0 00-.56 0l-5.43 2.7-.06-.01a.05.05 0 01-.02-.06l1.64-5.94a.62.62 0 00-.17-.6l-4.42-4.4a.05.05 0 01.03-.07l5.38-.54c.21-.02.4-.15.5-.34l2.78-5.52c0-.01.03-.03.05-.03z"
          fill="currentColor"
          fill-rule="evenodd"
          transform="scale(Infinity, Infinity)"
        />
      </svg>
    </button>
    <button
      class="zep-button zep-button--icon zep-button--secondary"
    >
      <svg
        aria-hidden="true"
        class="zep-button__icon"
        data-testid="ZepIconEssentialStarIcon"
        focusable="false"
        height="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M11.85 2.01a1.3 1.3 0 00-1.01.7L8.2 7.94l-5.03.5c-.5.04-.93.37-1.1.85l-.04.13c-.11.44.02.9.35 1.23l4.16 4.12L5 20.36c-.13.5.04 1.02.44 1.35l.11.08c.39.25.88.28 1.3.07L12 19.31l5.16 2.56c.46.22 1 .16 1.4-.16l.1-.1c.33-.32.46-.8.34-1.24l-1.54-5.6 4.16-4.13c.36-.35.48-.88.3-1.36l-.05-.12c-.2-.41-.6-.69-1.06-.73l-5.01-.5-2.63-5.21A1.3 1.3 0 0012 2l-.15.01zM12 3.25c.02 0 .04.02.05.03l2.79 5.53c.1.19.28.32.5.34l5.36.53c.03 0 .04.01.05.03l-.01.05-4.42 4.37a.62.62 0 00-.16.61l1.64 5.95c0 .02 0 .04-.02.05a.05.05 0 01-.06 0l-5.44-2.69a.62.62 0 00-.56 0l-5.43 2.7-.06-.01a.05.05 0 01-.02-.06l1.64-5.94a.62.62 0 00-.17-.6l-4.42-4.4a.05.05 0 01.03-.07l5.38-.54c.21-.02.4-.15.5-.34l2.78-5.52c0-.01.03-.03.05-.03z"
          fill="currentColor"
          fill-rule="evenodd"
          transform="scale(Infinity, Infinity)"
        />
      </svg>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
